plugins/nordic-hid/fu-nordic-hid-archive.c

```diff
@@ -11,15 +11,198 @@
+			g_set_error_literal(error,
+					    FWUPD_ERROR,
+					    FWUPD_ERROR_INVALID_FILE,
+					    "manifest invalid as has no board information");
+			return NULL;
+		}
+		return g_strdup(board_split[0]);
+	}
+
+	/* duplicate string for consistent memory management */
+	return g_strdup(board_name_readout);
+}
+
+static gint
+fu_nordic_hid_archive_parse_file_get_flash_area_id_v1(JsonObject *obj,
+						      const gchar *bootloader_name,
+						      guint files_cnt,
+						      GError **error)
+{
+	const gchar *image_idx_str = NULL;
+	const gchar *slot_str = NULL;
+	gint64 image_idx = -1;
+	gint64 slot = -1;
+
+	/* for MCUboot bootloader with swap, if only a single image is available,
+	 * the "image_index" and "slot" properties may be omitted
+	 */
+	if (g_strcmp0(bootloader_name, "MCUBOOT") == 0 && files_cnt == 1) {
+		return 0;
+	}
+
+	if (!json_object_has_member(obj, "image_index")) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "missing image_index property");
+		return -1;
+	}
+	image_idx_str = json_object_get_string_member(obj, "image_index");
+	if (image_idx_str == NULL) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "missing image_index property");
+		return -1;
+	}
+	if (!fu_strtoll(image_idx_str,
+			&image_idx,
+			G_MININT64,
+			G_MAXINT64,
+			FU_INTEGER_BASE_AUTO,
+			error)) {
+		g_prefix_error(error, "fu_strtoll failed for image_index:");
+		return -1;
+	}
+
+	if (!json_object_has_member(obj, "slot")) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "missing slot property");
+		return -1;
+	}
+	slot_str = json_object_get_string_member(obj, "slot");
+	if (slot_str == NULL) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "missing slot property");
+		return -1;
+	}
+	if (!fu_strtoll(slot_str, &slot, G_MININT64, G_MAXINT64, FU_INTEGER_BASE_AUTO, error)) {
+		g_prefix_error(error, "fu_strtoll failed for slot:");
+		return -1;
+	}
+
+	if (image_idx != 0) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "unsupported image_index property");
+		return -1;
+	}
+	if (slot != 0 && slot != 1) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "unsupported slot property");
+		return -1;
+	}
+
+	return slot;
+}
+
+static gint
+fu_nordic_hid_archive_parse_file_get_flash_area_id(JsonObject *obj,
+						   gint64 manifest_ver,
+						   guint file_idx,
+						   const gchar *bootloader_name,
+						   guint files_cnt,
+						   GError **error)
+{
+	/* for manifest version 0, the images are expected to be listed in strict order */
+	if (manifest_ver == 0)
+		return file_idx;
+	if (manifest_ver == 1)
+		return fu_nordic_hid_archive_parse_file_get_flash_area_id_v1(obj,
+									     bootloader_name,
+									     files_cnt,
+									     error);
```

Comment:
needs {} here

Reply:
Are you sure the `{}` is necessary here? Keep in mind that it's just a single multi-argument function call.

Anyway, I will add brackets if it was confusing :)

Reply:
Ohh it's not *needed* -- I think it just looks weird without one :)

Reply:
Added brackets :)